compiler/il/Aliases.cpp

```diff
@@ -647,6 +647,22 @@ OMR::SymbolReference::getUseDefAliasesBV(bool isDirectCall, bool includeGCSafePo
             aliases->set(symRefTab->getArrayShadowIndex(_symbol->getDataType().scalarToVector()));
             }
 
+         if (_symbol->isArrayShadowSymbol() &&
+             !symRefTab->aliasBuilder.immutableArrayElementSymRefs().isEmpty())
```

Comment:
Because a usual array shadow shares its `Symbol` with the corresponding immutable array shadow, and both have `reallySharesSymbol` set, I think these aliases should already be included via `setSharedShadowAliases()`

Reply:
Putting our discussion here:
`setSharedShadowAliases` only search for symrefs generated for non-array shadows with the same symbol, and `reallySharesSymbol` is needed for other types of aliasing such as `canKill`, so we have to keep `reallySharesSymbol` flag set as well as the loop iterating through `aliasBuilder.arrayElementSymRefs` to look for the alias set.